[Dev] Migrated to Biome 2.2.3, added more rules (#6259)
pagefaultgames/pokerogue: docs/localization.md

# Localization 101

> PokéRogue's localization team puts immense effort into making the game accessible around the world, supporting over 12 different languages at the time of writing this document.

Reviewer: ```suggestion
PokéRogue's localization team puts immense effort into making the game accessible around the world, supporting over 12 different languages at the time of writing this document. \
```
Since this had a double space, presumably it was to create a linebreak.

The following command will initialize your branch's locales repository and update its HEAD:
```bash

> pnpm update:locales

Reviewer: ```suggestion
pnpm update-locales
```

    ```

# Submitting Locales Changes
If you have a feature or enhancement that requires additions or changes to in-game text, you will need to make a fork of the `pokerogue-locales` repo and submit your text changes as a pull request _in addition_ to your pull request to the main project. \
Since these two PRs aren't _technically_ linked, it's important to coordinate with the Translation Team to ensure that both PRs are integrated safely into the project.

> [!CAUTION]
> **DO NOT HARDCODE PLAYER-FACING TEXT INTO THE CODE!**

## Making Changes

One perk of submodules is you don't actually _need_ to clone the locales repository to start contributing - `git` already does that for you on initialization.

Given `pokerogue-locales` is a full-fledged `git` repository _inside_ `pokerogue`, making changes is roughly the same as normal, merely using `public/locales` as your root directory.

> [!WARNING]
> Make sure to checkout or rebase onto `upstream/main` (`pnpm update-locales:remote`) **BEFORE** creating a locales PR!

> > The checked-out commit is based on the superproject's SHA-1 by default, so hastily making changes may see you basing your commits on last week 's `HEAD`.

Reviewer: ```suggestion
> The checked-out commit is based on the superproject's SHA-1 by default, so hastily making changes may see you basing your commits on last week's `HEAD`.
```

<!-- Remember to update this everytime the head of translation changes! -->
> [!IMPORTANT]

> > The current Head of Translation is:

Reviewer: ```suggestion
> The current Head of Translation is: \
```
You still need this (unless you want to make the line below into a list instead).
